lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
-  end
-
-  def add_services(service)
-    @services << service
-  end
-end
+
+    attr_reader :name,
+                :address, 
+                :phone, 
+                :services,
+                :registered_vehicles,
+                :collected_fees
+
+    def initialize(facility_info)
+        @name = facility_info[:name] 
+        @address = facility_info[:address]
+        @phone = facility_info[:phone]
+        @services = []
+        @registered_vehicles = []
+        @collected_fees = 0
+    end
+
+    # Add Service
```

Comment:
No need to keep these types of comments in a final draft. Ruby code should be mostly self-explanatory.